src/test/java/io/sniffy/socket/EchoServerRule.java

```diff
@@ -0,0 +1,190 @@
+package io.sniffy.socket;
+
+import org.junit.rules.ExternalResource;
+
+import java.io.*;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+public class EchoServerRule extends ExternalResource implements Runnable {
+
+    private final Thread thread = new Thread(this);
+
+    private final List<Thread> socketThreads = new ArrayList<>();
+    private final List<Socket> sockets = new ArrayList<>();
+
+    private int boundPort = 10000;
+    private ServerSocket serverSocket;
+
+    public int getBoundPort() {
+        return boundPort;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+
+        for (int i = 0; i < 10; i++, boundPort++) {
+            try {
+                serverSocket = new ServerSocket(boundPort, 50, InetAddress.getByName(null));
+                break;
+            } catch (IOException e) {
+                try {
+                    serverSocket.close();
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                }
+            }
+        }
+
+        if (null == serverSocket) {
+            throw new IOException("Failed to find an available port");
+        }
+
+        thread.start();
+
+    }
+
+    private CountDownLatch countDownLatch = new CountDownLatch(2);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/sniffy/sniffy/file/1633591209/issues/source?bid=3005960&fileBranchId=3034483#l52)